src/renderer/hooks/observable.ts

```diff
@@ -0,0 +1,31 @@
+import { useEffect, useState } from 'react';
+import { Observable } from 'rxjs';
+
+export function useObservable<T>(observable: Observable<T>, handler: (value: T) => void) {
+  useEffect(() => {
+    const subscription = observable.subscribe(handler);
+
+    return () => {
+      subscription.unsubscribe();
+    };
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Are there unintended consequences of including `observable` and `handler` in the useEffect dependency array?

Ignoring that ESLint rule is often a code smell and in this case I'd think we'd _want_ to re-run the effect if the reference to the observable or handler function changed.